File: Contests/RATED/CODEFORCES (RATED)/2_27_2024 (DIV 3)/C_Turtle_Fingers_Count_the_Values_of_k.cpp
#include <bits/stdc++.h>

using namespace std;

void divide(unordered_set<long long> &count, long long n, long long a, long long b){
    count.insert(n);
    // cout<<n<<" ";
    if (n%a==0){
        divide(count, n/a, a, b);
    }
    if (n%b==0){
        divide(count, n/b, a, b);
    }
}

int main() {
    int t;

    cin >> t;

    while (t--){
        long long l, a, b;
        cin >> a >> b >> l;
        unordered_set<long long> count;
        divide(count, l, a, b);
        cout << count.size() << endl;
    }
}

File: Contests/RATED/CODEFORCES (RATED)/2_27_2024 (DIV 3)/C_Turtle_Fingers_Count_the_Values_of_k.py
count = set()

def divide(n, a, b):
    count.add(n)
    if n > 1 and n % a == 0:
        divide(n//a, a, b)
    if n > 1 and n % b == 0:
        divide(n//b, a, b)

for i in range(int(input())):
    count = set()
    a,b,l = map(int, input().split())


    if l % a != 0 and l % b != 0:
        print(1)
    else:
        divide(l, a, b)
        print(len(count))

    
